pkgs/servers/scylladb/default.nix

```diff
@@ -0,0 +1,100 @@
+{ stdenv,
+  fetchgit,
+  python3Packages,
+  python3,
+  pkgconfig,
+  gcc8Stdenv,
+  boost,
+  git,
+  systemd,
+  gnutls,
+  cmake,
+  makeWrapper,
+  ninja,
+  ragel,
+  hwloc,
+  jsoncpp,
+  libantlr3cpp,
+  antlr3,
+  numactl,
+  protobuf,
+  cryptopp,
+  libxfs,
+  libyamlcpp,
+  libsystemtap,
+  lksctp-tools,
+  lz4,
+  libxml2,
+  zlib,
+  libpciaccess,
+  snappy,
+  libtool,
+  thrift-0_10
+}:
+gcc8Stdenv.mkDerivation rec {
+  pname = "scylladb";
+  version = "3.0.5";
+
+  src = fetchgit {
+    url = "https://github.com/scylladb/scylla.git";
+    rev = "403f66ecad6bc773712c69c4a80ebd172eb48b13";
+    sha256 = "14mg0kzpkrxvwqyiy19ndy4rsc7s5gnv2gwd3xdwm1lx1ln8ywsi";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./seastar-configure-script-paths.patch ];
+
+  nativeBuildInputs = [
+   pkgconfig
+   cmake
+   makeWrapper
+   ninja
+  ];
+
+  buildInputs = [
+   python3Packages.pyparsing
+   python3
+   boost
+   git
+   systemd
+   gnutls
+   ragel
+   hwloc
+   jsoncpp
+   libantlr3cpp
+   numactl
+   antlr3
+   protobuf
+   cryptopp
+   libxfs
+   libyamlcpp
+   libsystemtap
+   lksctp-tools
+   lz4
+   libxml2
+   zlib
+   libpciaccess
+   snappy
+   libtool
+   thrift-0_10
+  ];
+
+  configurePhase = ''
+     patchShebangs ./configure.py
+    ./configure.py --mode=release
```

Comment:
How about using `configureFlags` instead.

Reply:
Great point! 

What I'm trying to do, instead of overriding the `configurePhase`, is this:

```
  configureScript = "./configure.py";
  configureFlags = [ "--mode=release" ];
  dontAddPrefix = true;
```

However for some reason, whenever `cmake` is available as a `nativeBuildInput`, it ignores my overrides and tries to use `cmake` instead.

Reply:
To be more precise, it adds a `fixing cmake files...` files, which I can't seam to turn off and somehow leads to this:

```
configuring
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/gccy6z2y06s0k8z
k743rar66438v0y7q-scylladb-3.0.5/include -DCMAKE_INSTALL_LIBDIR=/nix/store/gccy6z2y06s0k8zk743rar66438v0y7q-scylladb-3.0.5/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/gccy6z2y06s0k8zk743rar66438v0y7q-scylladb-3.0.5/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_DEPLOYMENT_TARGET= -
DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/6k95q6vvnmspf3c0nm8q55sixsvmn9nc-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/6k95q6vvnmspf3c0nm8q55sixsvmn9nc-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/6k95q6vvnmspf3c0nm8q55sixsvmn9nc-binutils-
2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++  
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /nix/store/lfhwv8kc8c94ix3y7bn34zz4k1c4zgha-gcc-wrapper-8.3.0/bin/gcc
-- Check for working C compiler: /nix/store/lfhwv8kc8c94ix3y7bn34zz4k1c4zgha-gcc-wrapper-8.3.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /nix/store/lfhwv8kc8c94ix3y7bn34zz4k1c4zgha-gcc-wrapper-8.3.0/bin/g++
-- Check for working CXX compiler: /nix/store/lfhwv8kc8c94ix3y7bn34zz4k1c4zgha-gcc-wrapper-8.3.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:9 (message):
  This CMakeLists.txt file is only valid for use in IDEs, please define
  FOR_IDE to acknowledge this.


-- Configuring incomplete, errors occurred!

```